src/app/api/senior/[id]/route.ts

```diff
@@ -106,13 +106,14 @@ export const PATCH = withSessionAndRole(
       );
       const studentIDsToAdd = seniorBody.StudentIDs;
 
+      const { StudentIDs: _, ...other } = seniorBody;
```

Comment:
This actually was a bug we didn't notice before. Without destructuring `StudentIDs`, the relation on `User` side is not updated.